docker/bash.sh

```diff
@@ -75,6 +75,27 @@ else
     CI_PY_ENV=""
 fi
 
+if [[ "${DOCKER_IMAGE_NAME}" == *"demo_vitis_ai"* && -d "/dev/shm" && -d "/opt/xilinx/dsa" && -d "/opt/xilinx/overlaybins" ]]; then
+    WORKSPACE_VOLUMES="-v /dev/shm:/dev/shm -v /opt/xilinx/dsa:/opt/xilinx/dsa -v /opt/xilinx/overlaybins:/opt/xilinx/overlaybins"
```

Comment:
This seems to be too xilinx specific. Is it possible to do it in another way ousdier docker/bash? e.g. like the runtime case in nvidia

Reply:
The check is indeed Xilinx specific but there are also NVIDIA and GPU specific checks (https://github.com/apache/incubator-tvm/blob/d4bfcd4c4f8861f6716406ebae391f9936434c4e/docker/bash.sh#L54) in the bash.sh so I don't see how this differs. And this is the usual way in which we setup the docker environment with Vitis-AI. Also, the change shouldn't affect the other docker environments.

Reply:
OK. Please add some comments on top of the if section to describe the set of env being setup. We will need to think about cleanly organize the code as we start to add support to more devices, this could be a good starting pt.

Thanks @jtuyls

Reply:
One option - out of scope for this specific patch: have a 1:1 function that will do "all that is special about dealing with this specific image."

In this case, `bash.sh` will still be self contained, but we know that some logic (as a function) is only executed if the image requires it.

```
enable_other_custom_image()
{
  # do all that is requested to enable this image
}

enable_vitis_ai()
{
  # do all that is requested to enable this image
}

# we only add entries here if there is something special to do about a specific image
case "${DOCKER_IMAGE_NAME}" in
  "demo_vitis_ai")
    enable_vitis_ai()
    ;;
  "other_custom_image")
    enable_other_custom_image()
    ;;
  ... 
esac 
```

Reply:
I like the proposal of @leandron. Would improve clarity in the bash.sh for multiple devices. We could additionally standardize on variables like $VITIS_AI_FLAGS, etc to be edited in the function and to be added to the `docker run` command.

@leandron You specifically want to leave this out of this PR as we could easily already write our Vitis-AI changes in this format?

Reply:
@jtuyls I feel it needs a bit more discussion so I wouldn't block this PR merge for that. Despite feeling the idea is appropriate, I think we need to collect feedback in a specific PR.